### PR TITLE
fixes #4375: provisioning no longer loops when using oVirt + compute profile

### PR DESCRIPTION
**Pull request by dmitri-d**

### CODE REVIEW

**Comment by domcleal** [comment]
Any chance of a test in this area?  We have some around the population of compute attributes from a profile, circa line 1275 in the host unit test.

**Comment by dmitri-d** [comment]
I was so happy I found the problem, I forgot that I could also add a test. Yes, will do!

File: app/models/host/managed.rb [hunk 1]
```diff
@@ -150,7 +150,7 @@ class Jail < ::Safemode::Jail
     validates :provision_method, :inclusion => {:in => PROVISION_METHODS, :message => N_('is unknown')}, :if => Proc.new {|host| host.managed?}
     validates :medium_id, :presence => true, :if => Proc.new { |host| host.validate_media? }
     validate :provision_method_in_capabilities
-    before_validation :set_compute_attributes, :only => :create
+    before_validation :set_compute_attributes, :on => :create
```
Comment (ohadlevy) [inline app/models/host/managed.rb:153]: good catch! too bad rails did not complain initially :)

**Comment by lzap** [comment]
Nice one, and it looks like this was the only occurance...

```
[lzap@lzapx foreman]$ ack before_validation | grep only
app/models/host/managed.rb:153:    before_validation :set_compute_attributes, :only => :create
```

**Comment by dmitri-d** [comment]
@domcleal: now with tests.

**Comment by domcleal** [comment]
You should probably copy what @ares did here to fix the rubocop error: https://github.com/ares/foreman/commit/87ba089ce7635cc287a41001fb8dc4a8721781d1

**Comment by dmitri-d** [comment]
[test]

File: test/unit/host_test.rb [hunk 2]
```diff
@@ -112,6 +112,20 @@ class HostTest < ActiveSupport::TestCase
     assert_equal "my.host.company.com", host.name
   end
 
+  test "sets compute attributes on create" do
+    Host.any_instance.expects(:set_compute_attributes).at_most_once.returns(true)
```
Comment (GregSutcliffe) [inline test/unit/host_test.rb]: As discussed in IRC, this should be .once, not .at_most_once.

**Comment by dmitri-d** [comment]
@GregSutcliffe, @domcleal: fixed the expectation in the test.

**Comment by GregSutcliffe** [comment]
Merged as b6ad9fc, thanks!

**Pull request closed without merging**